x-pack/plugins/lens/public/indexpattern_datasource/loader.ts

```diff
@@ -54,20 +45,63 @@ export async function loadIndexPatterns({
     return cache;
   }
 
-  const resp = await savedObjectsClient.bulkGet(
-    missingIds.map((id) => ({ id, type: 'index-pattern' }))
-  );
+  const indexPatterns = await Promise.all(missingIds.map((id) => indexPatternsService.get(id)));
+  const indexPatternsObject = indexPatterns.reduce(
+    (acc, indexPattern) => {
+      const newFields = (indexPattern.fields as IndexPatternField[])
+        .filter(
+          (field) =>
+            !indexPatternsUtils.isNestedField(field) && (!!field.aggregatable || !!field.scripted)
+        )
+        .map(
+          (field): IndexPatternField => ({
+            name: field.name,
+            displayName: field.displayName,
+            type: field.type,
+            esTypes: field.esTypes,
+            aggregatable: field.aggregatable,
+            scripted: field.scripted,
+            searchable: field.searchable,
+            aggregationRestrictions: field.aggregationRestrictions,
```

Comment:
If you remove the case from above, this line will error because the field of an `IndexPattern` instance doesn't have `aggregationRestrictions`. You can simply set this to an empty object, because to logic below will map the restrictions into the fields.